src/QtLocationPlugin/GenericMapProvider.h

```diff
@@ -60,6 +60,17 @@ class JapanReliefMapProvider : public MapProvider {
     QString _getURL(const int x, const int y, const int zoom, QNetworkAccessManager* networkManager) override;
 };
 
+class CustomURLMapProvider : public MapProvider {
+    Q_OBJECT
+  public:
+    CustomURLMapProvider(QObject* parent = nullptr)
+        : MapProvider(QStringLiteral(""), QStringLiteral(""),
```

Comment:
Use default constructor `QStringLiteral()` to avoid the creation of an empty string

Reply:
@patrickelectric I've moved away from trying to use QStringLiteral everywhere and just use naked strings most places now. The reason being is that I think the impact to readability of the code is higher than the benefit of the unmeasurable perf hit of the on the fly generation/allocation.